src/ucp/core/ucp_mm.c

```diff
@@ -106,6 +106,30 @@ static ucs_status_t ucp_memh_reg_mds(ucp_context_h context, ucp_mem_h memh,
     return UCS_OK;
 }
 
+
+static ucs_status_t ucp_memh_detect_mds(ucp_context_h context, ucp_mem_h memh)
+{
+    ucs_status_t status;
+    unsigned md_index;
+
+    for (md_index = 0; md_index < context->num_mds; ++md_index) {
+        int is_cuda_device = 0;
+        status = uct_md_mem_detect(context->tl_mds[md_index].md, memh, memh->address,
+                                    memh->length, &is_cuda_device);
+        if (status != UCS_OK) {
+            return status;
+        }
+ 
+        if (is_cuda_device)
+            memh->md_type |= UCP_MD_FLAG_DEVICE_CUDA;
+        else
+            memh->md_type |= UCP_MD_FLAG_DEVICE_HOST;
+    }
+
+    return UCS_OK;
+}
+
+
```

Comment:
flags should be UCT flags, And i this is not exactly what we discussed yesterday.  this is not portable and ucx standard